### DISCUSSION TITLE
C# Support

### DISCUSSION THREAD

**Discussion by cb-eli** [microsoft/TypeChat#60] [General]
Is there a plan to release this library for C#?
Would be happy to start using it.

Thanks.

**Comment by DanielRosenwasser** [comment]
It's absolutely something that we've been discussing and exploring. We're trying to understand what the story might be for other languages over time - I spoke about this a little over at https://github.com/microsoft/TypeChat/discussions/35#discussioncomment-6513774.

**Comment by hermanussen** [comment]
I published a port of TypeChat in C# [here](https://github.com/hermanussen/TypeChatSharp). Of course it's unofficial and somewhat experimental at this point. CC @DanielRosenwasser

**Comment by 0xDing** [comment]
Our open-source project,[Flappy](https://github.com/pleisto/flappy), includes a C# implementation and features functionalities similar to Typechat and Code Interpreter.

**Comment by nulllogicone** [comment]
I don't know how the projects are related but found this C# implementation: https://github.com/microsoft/typechat.net
